rpm_s3_mirror/mirror.py

```diff
@@ -69,12 +71,17 @@ def sync(self):
         self.stats.gauge(metric="s3_mirror_sync_seconds_total", value=time.monotonic() - start)
         return sync_success
 
-    def _sync_repository(self, upstream_repository):
+    def _sync_repository(self, upstream_repository: RPMRepository):
         mirror_start = time.monotonic()
         update_time = now()
-        upstream_metadata = upstream_repository.parse_metadata()
-
-        mirror_repository = RPMRepository(base_url=self._build_s3_url(upstream_repository.base_url))
+        upstream_metadata: Metadata = upstream_repository.parse_metadata()
+        # Cache the repomd.xml when 1st reading from upstream repo.
+        base_url = upstream_repository.base_url
+        url = f"{base_url}repodata/repomd.xml"
+        cache_xml_path = download_file(temp_dir="/var/tmp", url=url, session=self.session)
```

Comment:
I reckon it's better to use a TemporaryDirectory with `prefix=self.config.scratch_dir` to store the cached file and handle cleaning it up afterwards. IIRC currently we do sync things sequentially but it can't hurt to play it safe. Also means you don't have to worry about the cleanup manually in `S3.overwrite_repomd`.

It is also a bit cleaner as it would be surprising if ever `overwrite_repomd` was used from somewhere else and then you discover that it also deletes files.